### PR TITLE
use resource name instead of Id in param when it can take both name a…

### PR DESCRIPTION
**Pull request by gauravsaralMs**
…nd id

### CODE REVIEW

File: azure-devops/azext_devops/dev/repos/repository.py [hunk 1]
```diff
@@ -95,12 +93,10 @@ def list_repos(devops_organization=None, project=None, detect=None):
...
 
 
-def show_repo(repo_id=None, name=None, devops_organization=None, project=None, detect=None, open_browser=False):
+def show_repo(repo, devops_organization=None, project=None, detect=None, open_browser=False):
     """Get the details of a Git repository.
-    :param repo_id: ID of the repository. Required if --name is not specified.
-    :type repo_id: str
-    :param name: Name of the repository. Ignored if --id is specified.
-    :type name: str
+    :param repo: ID or name of the repository. Required if --name is not specified.
```
Comment (ishitam8) [inline azure-devops/azext_devops/dev/repos/repository.py]: --name was removed. Pls update the help text.

File: azure-devops/azext_devops/dev/pipelines/arguments.py [hunk 2]
```diff
@@ -34,12 +34,6 @@ def load_build_arguments(self, _):
         context.argument('definition_id', type=int)
         context.argument('variables', nargs='*')
 
-    with self.argument_context('pipelines build show') as context:
-        context.argument('build_id', options_list=('--build-id', '--id'), type=int)
-
-    with self.argument_context('pipelines build definition show') as context:
-        context.argument('definition_id', options_list=('--definition-id', '--id'), type=int)
-
```
Comment (ishitam8) [inline azure-devops/azext_devops/dev/pipelines/arguments.py]: Couldn't comment at unchanged line number. 
task_id should be renamed to 'id' as well

File: azure-devops/azext_devops/dev/repos/repository.py (hunk 1 above)
Comment (gauravsaralMs) [reply azure-devops/azext_devops/dev/repos/repository.py]: done

File: azure-devops/azext_devops/dev/team/service_endpoint.py [hunk 3]
```diff
@@ -41,10 +41,10 @@ def list_service_endpoints(devops_organization=None, project=None, detect=None):
         raise CLIError(ex)
 
 
-def show_service_endpoint(id, devops_organization=None, project=None, detect=None):  # pylint: disable=redefined-builtin
+def show_service_endpoint(service_endpoint, devops_organization=None, project=None, detect=None):
     """Get the details of a service endpoint.
-    :param id: ID of the service endpoint.
-    :type id: str
+    :param service_endpoint: ID of the service endpoint.
```
[the remark this answers is not in the document]
Comment (gauravsaralMs) [reply azure-devops/azext_devops/dev/team/service_endpoint.py]: done

**Pull request merged**